meetings/Meetings.js

```diff
@@ -0,0 +1,358 @@
+/** Script to generate various meeting reports.
+ *
+ * NOTE: You may install periodic triggers to keep these reports up-to-date in the output sheets.
+ */
+
+/** The prefix for properties specific to this script in the project. */
+const PROPERTY_PREFIX = 'Meetings.';
+
+/** Personio clientId and clientSecret, separated by '|'. */
+const PERSONIO_TOKEN_KEY = PROPERTY_PREFIX + 'personioToken';
+
+/** Service account credentials (in JSON format, as downloaded from Google Management Console). */
+const SERVICE_ACCOUNT_CREDENTIALS_KEY = PROPERTY_PREFIX + 'serviceAccountCredentials';
+
+/** Filter for allowed domains (to avoid working and failing on users present on foreign domains). */
+const ALLOWED_DOMAINS_KEY = PROPERTY_PREFIX + 'allowedDomains';
+
+/** White-list to restrict synchronization to a few tester email accounts.
+ *
+ * Must be one email or a comma separated list.
+ *
+ * Default: null or empty
+ */
+const EMAIL_WHITELIST_KEY = PROPERTY_PREFIX + 'emailWhiteList';
+
+/** Lookahead days for event/time-off synchronization.
+ *
+ * Default: 6 * 30 days, should scale up to ~18 months
+ */
+const LOOKAHEAD_DAYS_KEY = PROPERTY_PREFIX + 'lookaheadDays';
+
+/** Lookback days for event/time-off synchronization.
+ *
+ * Default: 30 days, should scale up to ~6 months, avoid changing events too far back
+ */
+const LOOKBACK_DAYS_KEY = PROPERTY_PREFIX + 'lookbackDays';
+
+/** ID of sheet to write reports to. */
+const REPORT_SHEET_KEY = PROPERTY_PREFIX + 'reportSheet';
+
+
+/** Build meetings statistic per employee and summarized. */
+async function listMeetingStatistic() {
+
+    const weeks = 4;   // 4 weeks back from now
+
+    const now = new Date();
+    const weekBounds = [];
+    for (let week = weeks - 1; week >= 0; --week) {
+        weekBounds.push({
+            start: new Date(new Date().setDate(now.getDate() - ((week + 1) * 7))),
+            end: new Date(new Date().setDate(now.getDate() - (week * 7)))
+        });
+    }
+
+    const getWeekIndex = (startDate, endDate) => {
+        for (let i = 0; i < weekBounds.length; ++i) {
+            if (+startDate >= +weekBounds[i].start && +startDate <= +weekBounds[i].end) {
+                return i;
+            }
+        }
+        for (let i = 0; i < weekBounds.length; ++i) {
+            if (+endDate >= +weekBounds[i].start && +endDate <= +weekBounds[i].end) {
+                return i;
+            }
+        }
+        return null;
+    };
+
+    const allowedDomains = (getScriptProperties_().getProperty(ALLOWED_DOMAINS_KEY) || '')
+        .split(',')
+        .map(d => d.trim());
+
+    const employeeStats = {};
+    try {
+        let haveEmployees = false;
+        await visitEvents_((event, employee, employees, calendar, personio) => {
```

Comment:
So the code in the following lines here is the "visitor" function which receives each event from `visitEvents_` as input, right?

I think some comments would help as the flow is not super obvious, it gets clear but only after engaging some brain cells ;)

Reply:
I added comments.

My IDE displays/tags function parameter names, thus this issue didn't occur to me:
![grafik](https://github.com/giantswarm/gapps-automation/assets/18163495/7d5d27f1-10fa-4ebb-a2ac-229bf7411a4a)